modules/fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/fs2/Environment.scala

```diff
@@ -68,13 +70,14 @@ final case class Environment[F[_]](
   assetsState: Assets.State[F],
   blocker: Blocker,
   source: RawSource[F],
-  good: ByteSink[F],
-  pii: Option[ByteSink[F]],
+  good: AttributedByteSink[F],
+  pii: Option[AttributedByteSink[F]],
   bad: ByteSink[F],
   sentry: Option[SentryClient],
   metrics: Metrics[F],
   assetsUpdatePeriod: Option[FiniteDuration],
-  metricsReportPeriod: Option[FiniteDuration]
+  goodAttributes: Set[String],
```

Comment:
Do we have any validation logic, checking that `foobar` is not possible? I was thinking about switching to `Set[Field]` at first, but also feel uncomfortable about polluting Scala code with something reflection-related.

Reply:
I do not validate names, but I ignore invalid names.  [This is the relevant line](https://github.com/snowplow/enrich/blob/9bdd973092716e61d57fe77d034f03c95e0e41f6/modules/fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/fs2/Enrich.scala#L115)

Reply:
I have a slight preference towards failing on unexpected names - I think silent skip can lead to a situation where user has a typo and expects certain behavior, but not getting it. Also fail-fast is more consistent with enrichments parsing logic.